### ISSUE TITLE
Found array with 0 sample(s)

### ISSUE THREAD

**Issue by pjmnoble** [MaartenGr/BERTopic#1550]
So Bertopic rocks !
I am looking at a set of clinical notes each with an associated breed (of dog)
I'm running topics_per_class and get:

Found array with 0 sample(s) (shape=(0, 53236)) while a minimum of 1 is required by the normalize function

I assume a topic occurs with no entries for a given breed (80k documents, 200breeds) but have no idea how to preprocess the data or model to allow it to give me topics/class.

This works beautifully with a model that uses 1 million documents but I'm guessing empty matrices don't happen much then

Would love some help

**Comment by pjmnoble** [comment]
There are breeds with 1 document - there are 185 breeds in all . .  I could set some thresholds and aggregate rare breeds as a  breed called 'rare'

Name: bertopic
Version: 0.15.0
Name: umap-learn
Version: 0.5.4
Name: hdbscan
Version: 0.8.33

77875 documents in a data frame working_df
```
working_df = working_df.loc[working_df.item_text!='.']
narratives = working_df.item_text.apply(str).to_list()
breed = working_df.breed.to_list()
```

model-creation:
```
hdbscan_model = hdbscan.HDBSCAN(min_cluster_size=30,
                               metric='euclidean',
                               cluster_selection_method='eom',
                               prediction_data=True,
                               core_dist_n_jobs=1)  # TODO should prevent error

stop_words = list(text.ENGLISH_STOP_WORDS.union(['bhb','jsd','roxys','rgb','rwg','mc','ro','em','hm','hpc']))
vectorizer_model = CountVectorizer(stop_words=stop_words)

model = BERTopic(language="english", 
                 top_n_words=20, 
                 verbose=True, 
                 hdbscan_model=hdbscan_model,
                 nr_topics = 50,
                 vectorizer_model=vectorizer_model)
topics, probabilities = model.fit_transform(narratives)
```
calling topics_per_class
```
breed_based_topics = model.topics_per_class(docs=narratives, 
                                        classes=breed, 
                                        global_tuning=False)
```

**Comment by MaartenGr** [comment]
> There are breeds with 1 document - there are 185 breeds in all . . I could set some thresholds and aggregate rare breeds as a breed called 'rare'

I think that would definitely help and would be worthwhile to try out! 

The rest of the code seems alright. One tip though, check out the [best practices](https://maartengr.github.io/BERTopic/getting_started/best_practices/best_practices.html) page for a number of tricks that can easily create much better-looking topics.

**Comment by pjmnoble** [comment]
yep, had just started to do that but wanted to do a look see at what I was getting . .  already very good stuff

Thanks for your quick advice

**Comment by MaartenGr** [comment]
Cool, glad to hear that it worked out!
